docs/spec/spec.md

```diff
@@ -407,52 +407,74 @@ status:
 ## Container
 
 This is a
-[core.v1.Container](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#container-v1-core).
-Some fields are not allowed, such as name and lifecycle.
+[core.v1.Container](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#container-v1-core).
+Fields not specified in the container object below are disallowed. Examples of
+disallowed fields include `name` and `lifecycle`.
 
 This type is not used on its own but is found composed inside
 [Service](#service), [Configuration](#configuration), and [Revision](#revision).
 
+Items with `...` behave as specified in the Kubernetes API documentation unless
+otherwise noted.
+
 ```yaml
 container: # v1.Container
-  # image either provided as pre-built container, or built by Knative Serving from
-  # source. When built by knative, set to the same as build template, e.g.
+  args: [...] # Optional
+  command: [...] # Optional
+  env: ... # Optional
+  envFrom: ... # Optional
+
+  # image is either provided as pre-built container, or built by Knative Serving from
+  # source. When built by Knative, set to the same as build template, e.g.
   # build.template.arguments[_IMAGE], as the "promise" of a future build.
   # If buildRef is provided, it is expected that this image will be
   # present when the referenced build is complete.
-  image: gcr.io/...
-  command: ["run"]
-  args: []
-  env:
-    # list of environment vars
-    - name: FOO
-      value: bar
-    - name: HELLO
-      value: world
-    - ...
-
-  # Optional, only a single containerPort may be specified.
+  image: gcr.io/... # Required
+  imagePullPolicy: ... # Optional
+
+  # HTTPGetAction and TCPSocketAction are the supported probe options.
```

Comment:
🤔 We support exec as well, which is needed for GRPC probing: https://kubernetes.io/blog/2018/10/01/health-checking-grpc-servers-on-kubernetes/

Pretty sure we do, since I'm using it in korpc for the above.

Reply:
We shouldn't be supporting exec according to our runtime contract ( https://github.com/knative/serving/blob/master/docs/runtime-contract.md#meta-requests ).

We will need to reconcile this.